File: cpp/Problem118.cpp
/*
 * Problem: 118. Pascal's Triangle
 * URL: https://leetcode.com/problems/pascals-triangle/
 * Difficulty: Easy
 * Tags: Array, Dynamic Programming(DP)
 */

#include <vector>

using namespace std;

class Solution {
public:
	vector<vector<int>> generate(int numRows) {
        vector<vector<int>> results;
        for (int row = 0; row < numRows; row++) {
            results.push_back(vector<int>(row + 1, 1));
            for (int col = 1; col < row; col++) {
                results[row][col] = results[row - 1][col - 1] + results[row - 1][col];
            }
        }
        return results;
	}
};

File: python/Problem118.py
# Problem: 118. Pascal's Triangle
# URL: https://leetcode.com/problems/pascals-triangle/
# Difficulty: Easy
# Tags: Array, Dynamic Programming(DP)

from typing import List

class Solution:
  def generate(self, numRows: int) -> List[List[int]]:
    results: List[List[int]] = []
    for row in range(numRows):
      results.append([1] * (row + 1))
      for col in range(1, row):
        results[row][col] = results[row - 1][col - 1] + results[row - 1][col]
    return results